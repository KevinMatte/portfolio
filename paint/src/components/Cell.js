import React, {Component} from "react";
import PropTypes from "prop-types";

import TextField from "@material-ui/core/TextField";

class Cell extends Component {

    handleChangeEvent = (event) => {
        this.props.setValue(event.target.value);
    };

    render() {
        let {value, doEdit} = this.props;

        if (doEdit) {
            return (
                <TextField value={value} onChange={event => this.handleChangeEvent(event)}>
                </TextField>
            )
        } else {
            return <span>{value}</span>;
        }

    }
}

Cell.defaultProps = {
    doEdit: false,
};

Cell.propTypes = {
    doEdit: PropTypes.bool,
    value: PropTypes.any.isRequired,
    setValue: PropTypes.func,
};

export default Cell;



