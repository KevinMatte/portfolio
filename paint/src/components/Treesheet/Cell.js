import React from "react";
import PropTypes from "prop-types";

import TextField from "@material-ui/core/TextField";

function Cell(props) {
    let {value, setValue, doEdit, dataTestID} = props;

    let handleChangeEvent = (event) => {
        setValue(event.target.value);
    };

    if (doEdit) {
        return (
            <TextField data-testid={dataTestID} value={value} onChange={handleChangeEvent}>
            </TextField>
        )
    } else {
        return <span data-testid={dataTestID}>{value}</span>;
    }
}

Cell.defaultProps = {
    doEdit: false,
    dataTestID: "Cell",
};

Cell.propTypes = {
    doEdit: PropTypes.bool,
    value: PropTypes.any.isRequired,
    setValue: PropTypes.func,
};

export default Cell;



