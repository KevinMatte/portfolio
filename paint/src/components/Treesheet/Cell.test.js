// Cell.react.test.js
import React from 'react';
import {render, fireEvent, cleanup, waitForElement} from 'react-testing-library';
import 'jest-dom/extend-expect'
import Cell from './Cell';
// import renderer from 'react-test-renderer';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test('Cell: Edit mode: off', () => {
  let tree, value;

  let mockSetValue = jest.fn();
  value = "hello";
  const {container, getByTestId, asFragment} = render(
    <Cell doEdit={false} value={value} setValue={mockSetValue}>Facebook</Cell>,
  );

  tree = asFragment();
  expect(tree).toMatchSnapshot();

});

test('Cell: Edit mode: on', () => {
  let tree, value;
  let mockSetValue = jest.fn();
  value = "hello";
  const {container, getByTestId, asFragment} = render(
    <Cell doEdit={true} value={value} setValue={mockSetValue}>Facebook</Cell>,
  );

  tree = asFragment();
  expect(tree).toMatchSnapshot();

  let cell = getByTestId("Cell");
  let input = cell.querySelector('input');
  fireEvent.change(input, { target: { value: 'A' } });
  expect(mockSetValue.mock.calls.length).toBe(1);
  expect(mockSetValue.mock.calls[0][0]).toBe('A');

});
