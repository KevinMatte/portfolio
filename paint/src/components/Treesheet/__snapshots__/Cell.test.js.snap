// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell: Edit mode: off 1`] = `
<DocumentFragment>
  <span
    data-testid="Cell"
  >
    hello
  </span>
</DocumentFragment>
`;

exports[`Cell: Edit mode: on 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root-1"
    data-testid="Cell"
  >
    <div
      class="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input-28 MuiInput-input-13"
        type="text"
        value="hello"
      />
    </div>
  </div>
</DocumentFragment>
`;
